### PR TITLE
Build improvements, CJS/ESM @clerk/clerk-react

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

**Please review commit by commit**

- Resolve circular imports in clerk/shared
- New esm/cjs build pipeline for clerk/shared
- New esm/cjs build pipeline for clerk/clerk-react
- script loader cleanup
- introduce clerkJSVersion
- enable isolatedModules for clerk/nextjs
- improved ESM support for clerk/js (reduces the bundle size by ~10kb)

The build improvements also include these notable changes:
- better source map support, especially within the monorepo
- easier node_modules debugging as clerk-react and clerk-shared are no longer bundled into a single file
- better nextjs error logging as the packages are not minified by default

### CODE REVIEW

**Comment by nikosdouvlis** [comment]
!snapshot

**Comment by nikosdouvlis** [comment]
!snapshot

**Comment by nikosdouvlis** [comment]
!snapshot

**Comment by nikosdouvlis** [comment]
!snapshot

**Comment by nikosdouvlis** [comment]
!snapshot

**Comment by nikosdouvlis** [comment]
!snapshot

File: packages/react/src/polyfills.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Vite does not define `global` by default
+ * One workaround is to use the `define` config prop
+ * https://vitejs.dev/config/#define
+ * We are solving this in the SDK level to reduce setup steps.
+ */
+if (typeof global === 'undefined' && typeof window !== 'undefined' && !window.global) {
+  console.log('nikos2?');
```
Comment (dimkl) [inline packages/react/src/polyfills.ts:8]: ✂️

File: packages/shared/src/utils/loadScript.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const NO_SRC_ERROR = 'loadScript cannot be called without a src';
+
+type LoadScriptOptions = {
+  async?: boolean;
+  defer?: boolean;
+  crossOrigin?: 'anonymous' | 'use-credentials';
+  beforeLoad?: (script: HTMLScriptElement) => void;
+};
+
+export async function loadScript(src = '', opts: LoadScriptOptions): Promise<HTMLScriptElement> {
```
Comment (anagstef) [inline packages/shared/src/utils/loadScript.ts:11]: ❓ Should we add tests for this?

File: packages/react/src/utils/loadClerkJsScript.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { addClerkPrefix, isValidProxyUrl, loadScript, parsePublishableKey, proxyUrlToAbsoluteURL } from '@clerk/shared';
```
Comment (anagstef) [inline packages/react/src/utils/loadClerkJsScript.ts:1]: ❓ Should we add tests for this file?

**Review by anagstef — APPROVED** [review]

**Review by anagstef — APPROVED** [review]

**Comment by nikosdouvlis** [comment]
!snapshot mytag

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**